File: baekjoon/java/25304. 영수증.java
import java.util.Scanner;

public class Main{

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int ptotal = sc.nextInt();
        int ntotal = sc.nextInt();
        int res = 0;
        for (int i=0;i<ntotal;i++){
            int price = sc.nextInt();
            int n = sc.nextInt();
            res += price*n;
        }
        if (ptotal == res) System.out.println("Yes");
        else System.out.println("No");
    }
}

File: baekjoon/cpp/25304.영수증.cpp
#include <iostream>
using namespace std;

int main(void){

    int ntotal, ptotal, n, price;
    int res = 0; 
    cin >> ptotal;
    cin >> ntotal;
    for (int i=0;i<ntotal;i++){
        cin >> price >> n;
        res += price*n;
    }    
    if (ptotal==res) cout << "Yes";
    else cout << "No";
    return 0;
}
